src/Description/SonataEnhancer.php

```diff
@@ -118,6 +120,29 @@ public function supports(PuliResource $resource)
         // sonata has dependency on ClassUtils so this is fine.
         $class = ClassUtils::getClass($payload);
 
-        return $this->pool->hasAdminByClass($class);
+        return null !== $this->getAdminByClass($class);
+
+    }
+
+    private function getAdminByClass($class)
+    {
+        if (isset($this->adminMap[$class])) {
+            return $this->adminMap[$class];
+        }
+
+        $_class = $class;
+        while ($_class && !$this->pool->hasAdminByClass($_class)) {
+            $_class = get_parent_class($_class);
+        }
+
+        if (false === $_class) {
+            $this->adminMap[$class] = null;
```

Comment:
or simply `$this->adminMap[$class] = $_class ? $this->pool->getAdminByClass($_class) : null; return $this->adminMap[$class];`